### PR TITLE
Building path with path's parameters suggestion

### PR DESCRIPTION
**Pull request by ngohado**
Sorry. I tried to implement building path with parameters without know we have already exist :( . But I still push and create PR to show you another view to create path with path's parameters.

### CODE REVIEW

File: data/lib/src/network/config/endpoint.dart [hunk 1]
```diff
@@ -44,7 +44,7 @@ class Endpoint {
 
   static final ServicePath sharedSpaces = ServicePath('/shared_spaces');
 
-  static final ServicePath autocomplete = ServicePath('/autocomplete');
+  static final ServicePath autocomplete = ServicePath(r'/autocomplete/$query');
```
Comment (hoangdat) [inline data/lib/src/network/config/endpoint.dart:47]: can you explain more clear? what is `r`

File: data/lib/src/network/linshare_http_client.dart [hunk 2]
```diff
@@ -119,7 +121,7 @@ class LinShareHttpClient {
   Future<List<AutoCompleteResult>> getSharingAutoComplete(
       AutoCompletePattern autoCompletePattern) async {
     final List resultJson = await _dioClient.get(Endpoint.autocomplete
-        .withPathParameter(autoCompletePattern.value)
+        .withPathParameters([PathParameter('query', autoCompletePattern.value)])
```
Comment (hoangdat) [inline data/lib/src/network/linshare_http_client.dart:124]: cool. I like it

File: data/test/network/endpoint_ext_test.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'package:domain/domain.dart';
+import 'package:data/src/network/config/endpoint.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('endpoint extension tests', () {
+    test('Service path build with path parameters', () {
```
Comment (hoangdat) [inline data/test/network/endpoint_ext_test.dart:7]: please test more cases, `null`, `exception`, ...

File: data/lib/data.dart [hunk 4]
```diff
@@ -31,6 +31,7 @@ export 'src/datasource_impl/autocomplete_datasource_impl.dart';
 
 // util
 export 'src/util/device_manager.dart';
+export 'src/network/config/endpoint.dart';
```
Comment (manhduc-nguyen) [inline data/lib/data.dart:34]: I don't think we need to export `endpoint.dart `

**Pull request closed without merging**